File: easy/longest_palindrome.rb
# https://leetcode.com/problems/longest-palindrome/

def longest_palindrome(s)
  result = 0
  character_count = Hash.new(0)

  s.each_char { |char| character_count[char] += 1 } 

  character_count.values.each do |count|
    if count.even?
      result += count
    elsif count.odd?
      result += count - 1
    end
  end

  result == s.length ? result : result + 1
end

File: easy/longest_palindrome.js
// https://leetcode.com/problems/longest-palindrome/submissions/

var longestPalindrome = function(s) {
  let result = 0;
  let character_count = {};

  s.split("").forEach(char => {
    if (character_count[char]) {
      character_count[char] += 1;
    } else {
      character_count[char] = 1;
    }
  });

  Object.values(character_count).forEach(value => {
    value % 2 === 0 ? result += value : result += value - 1;
  });

  return (result === s.length ? result : result + 1);
};